presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorConnector.java

```diff
@@ -167,12 +189,90 @@ public void testMaintenanceUnblockedOnStart()
         assertFalse(metadataDao.isMaintenanceBlockedLocked(tableId));
     }
 
+    @Test
+    public void tesTemporalShardSplit()
+            throws Exception
+    {
+        assertSplitShard(DATE, "2001-08-22", "2001-08-22", "UTC", 1);
+        // Date should not be affected by timezone
+        assertSplitShard(DATE, "2001-08-22", "2001-08-22", "America/Los_Angeles", 1);
+        // User timezone is UTC, while system shard split timezone is PST
+        assertSplitShard(TIMESTAMP, "2001-08-22 00:00:01.000", "2001-08-22 23:59:01.000", "UTC", 2);
+        // User timezone is PST, while system shard split timezone is PST
+        assertSplitShard(TIMESTAMP, "2001-08-22 00:00:01.000", "2001-08-22 23:59:01.000", "America/Los_Angeles", 1);
+    }
+
+    private void assertSplitShard(Type temporalType, String min, String max, String userTimeZone, int numSplit)
```

Comment:
I think we can do this test much simpler as a write query in `TestRaptorIntegrationSmokeTest` and a verification using the `$shard_uuid` column (see the bucket tests)

Reply:
Unfortunately, using IntegrationSmokeTest in this case is more difficult. First, changing timezone for different test is difficult because the session properties are shared between tests. Secondly, we validates result against H2 so under current implementation only UTC timezone will produce same result as H2.